src/net/if_.rs

```diff
@@ -17,3 +17,311 @@ pub fn if_nametoindex<P: ?Sized + NixPath>(name: &P) -> Result<c_uint> {
+
+        // bitmask: 0x80
+        /// No arp protocol, L2 destination address not set. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        IFF_NOARP;
+
+        // bitmask: 0x0100
+        /// Interface is in promiscuous mode. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        IFF_PROMISC;
+
+        // bitmask: 0x0200
+        /// Receive all multicast packets. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        IFF_ALLMULTI;
+
+        // bitmask: 0x0400
+        /// Master of a load balancing bundle. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+        IFF_MASTER;
+        /// tx hardware queue is full
+        #[cfg(target_os = "freebsd")]
+        IFF_DRV_OACTIVE;
+        /// transmission in progress
+        #[cfg(any(target_os = "bitrig",
+                  target_os = "macos",
+                  target_os = "netbsd",
+                  target_os = "openbsd",
+                  target_os = "ios"))]
+        IFF_OACTIVE;
+        /// Protocol code on board.
+        #[cfg(target_os = "solaris")]
+        IFF_INTELLIGENT;
+
+        // bitmask: 0x0800
+        /// Slave of a load balancing bundle. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+        IFF_SLAVE;
+        /// Can't hear own transmissions.
+        #[cfg(any(target_os = "bitrig",
+                  target_os = "dragonfly",
+                  target_os = "freebsd",
+                  target_os = "macos",
+                  target_os = "netbsd",
+                  target_os = "openbsd",
+                  target_os = "osx"))]
+        IFF_SIMPLEX;
+
+        // This has bitmask: 0x1000 on android, fuchsia, and linux
+        //              but  0x0800 on solaris
+        //              and  0x8000 on dragonfly, macos, netbsd, openbsd, and ios
+        /// Supports multicast. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        IFF_MULTICAST;
+
+        // bitmask: 0x1000
+        /// Per link layer defined bit.
+        #[cfg(any(target_os = "bitrig", target_os = "dragonfly", target_os = "freebsd", target_os = "macos", target_os = "netbsd", target_os = "openbsd", target_os = "ios"))]
+        IFF_LINK0;
+        /// Multicast using broadcast.
+        #[cfg(any(target_os = "solaris"))]
+        IFF_MULTI_BCAST;
+
+        // bitmask: 0x2000
+        /// Is able to select media type via ifmap. (see
+        /// [`netdevice(7)`](http://man7.org/linux/man-pages/man7/netdevice.7.html))
+        #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "fuchsia", target_os = "linux"))]
+        IFF_PORTSEL;
```

Comment:
This is what caused your build failure.  `IFF_PORTSEL` is not defined on FreeBSD.

Reply:
Ah, I see. That's clearly wrong. Corrected.